server/apps/microservices/src/processors/video-transcode.processor.ts

```diff
@@ -91,6 +92,13 @@ export class VideoTranscodeProcessor {
     if (longestVideoStream.codec_name !== config.ffmpeg.targetVideoCodec) {
       return this.runFFMPEGPipeLine(asset, savedEncodedPath);
     }
+
+    if (config.ffmpeg.transcode == TranscodePreset.OPTIMAL) {
+      const videoHeightThreshold = 1080;
+      if (!longestVideoStream.height || longestVideoStream.height > videoHeightThreshold) {
+        return this.runFFMPEGPipeLine(asset, savedEncodedPath);
+      }
+    }
```

Comment:
Thanks for the PR. A welcoming change, as I'm facing this issue myself. I do have a few notes/questions though:
1. 1080 is a pretty sensible value, it would be nice to make this configurable but I think it's fine for now
2. What if someone uploads a vertical video? Shouldn't it check if both the height or width exceed the threshold?
3. If the height is unavailable it will always run the transcoding pipeline. Is that intended behavior?

Also a general comment: please use strict equality checks (triple equals) whenever possible

Reply:
> * 1080 is a pretty sensible value, it would be nice to make this configurable but I think it's fine for now
> * What if someone uploads a vertical video? Shouldn't it check if both the height or width exceed the threshold?

I decided to adapt my logic from Google's Storage Saver plan where they do the following: `Videos higher than 1080p will be resized to high-definition 1080p`. I suppose that's enough to start with

> If the height is unavailable it will always run the transcoding pipeline. Is that intended behavior?

I can hardly imagine a video with height unavailable. Not sure but decided to transcode video in this case

> Also a general comment: please use strict equality checks (triple equals) whenever possible

Thanks, I will fix it

Reply:
Replaced loose equality checks with scrict ones while refactoring `runVideoEncode` method